packages/core/content-manager/server/src/services/utils/__tests__/validatable-fields-populate.test.ts

```diff
@@ -0,0 +1,173 @@
+import { getPopulateForValidation } from '../populate';
+
+describe('getPopulateForValidation', () => {
+  const fakeModels = {
+    empty: {
+      modelName: 'Fake empty model',
+      attributes: {},
+    },
+    scalarOnly: {
+      modelName: 'Fake scalar-only model',
+      attributes: {
+        title: { type: 'string', required: true },
+        description: { type: 'text', required: false },
+      },
+    },
+    componentWithRequiredFields: {
+      modelName: 'Fake component with required fields',
+      attributes: {
+        componentAttrName: {
+          type: 'component',
+          component: 'componentFields',
+        },
+      },
+    },
+    componentFields: {
+      modelName: 'Fake component fields',
+      attributes: {
+        subfield1: { type: 'string', required: true },
+        subfield2: { type: 'number', required: false },
+      },
+    },
+    componentWithoutRequiredFields: {
+      modelName: 'Fake component without required fields',
+      attributes: {
+        componentAttrName: {
+          type: 'component',
+          component: 'empty',
+        },
+      },
+    },
+  } as any;
+
+  beforeEach(() => {
+    global.strapi = {
+      getModel: jest.fn((uid) => fakeModels[uid]),
+    } as any;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('with empty model', () => {
+    const uid = 'empty';
+
+    const result = getPopulateForValidation(uid as any);
+
+    expect(result).toEqual({});
+  });
+
+  test('with scalar-only model', () => {
+    const uid = 'scalarOnly';
+
+    const result = getPopulateForValidation(uid as any);
+
+    expect(result).toEqual({
+      fields: ['title'], // Only scalar fields requiring validation
+    });
+  });
+
+  describe('components', () => {
+    test('with component model containing required fields', () => {
+      const uid = 'componentWithRequiredFields';
+
+      const result = getPopulateForValidation(uid as any);
+
+      expect(result).toEqual({
+        populate: {
+          componentAttrName: {
+            fields: ['subfield1'], // Only required fields in the component
+          },
+        },
+      });
+    });
```

Comment:
Sooo, I am trying to create a function to select fields inside components, but it turns out there is a bug that prevents this. Until we fix that this PR will be blocked 😢 cc @remidej @markkaylor 

We use those fields populated inside the "localizations" attribute when bulk publishing locales

Reply:
Hey Marc. So to merge your improvement we'd also need to update the bulk publish functionality? What's the bug exactly?

Reply:
1st issue that I am trying to fix: 
- the bulk publish UI needs required (and others) fields available inside `localizations` to know if a locale is publisheable or not

2nd bug I found while doing the 1st:
- we can not select fields inside component, I am not sure how we made it work in the first place (cc @jhoward1994 , maybe you can give us some hints to unblock this)
e.g of whats not possible:
```ts
populate: {
  compo: {
    fields: ['name'] // This fails
  },
}
```

Reply:
I am going to take a look at the failing API tests too

Reply:
For @jhoward1994 

the current getValidatableFieldsPopulate() function doesnt give a right output atm, for the address content type:

![image](https://github.com/user-attachments/assets/2dc8716c-896d-415b-8b0c-7e338572af1d)

It tries to populate non populatable fields (city, postal_code), which makes the document service validation fail.

I have tried to propose a different approach, but fields can not be selected in components at the moment